pkg/pillar/types/global.go

```diff
@@ -103,312 +106,410 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
+		return val.boolValue
+	} else {
+		return false
 	}
-	if newgc.VdiskGCTime < GlobalConfigMinimums.VdiskGCTime {
-		log.Warnf("Enforce minimum VdiskGCTime received %d; using %d",
-			newgc.VdiskGCTime, GlobalConfigMinimums.VdiskGCTime)
-		newgc.VdiskGCTime = GlobalConfigMinimums.VdiskGCTime
+}
+
+func (configPtr *ConfigItemValueMap) SetAgentSettingStringValue(key AgentSettingKey, agentName string, newValue string) error {
+	configItemValue := ConfigItemValue {
+		Key: string(key),
+		ItemType: ConfigItemTypeString,
+		stringValue: newValue,
 	}
-	if newgc.DownloadRetryTime < GlobalConfigMinimums.DownloadRetryTime {
-		log.Warnf("Enforce minimum DownloadRetryTime received %d; using %d",
-			newgc.DownloadRetryTime, GlobalConfigMinimums.DownloadRetryTime)
-		newgc.DownloadRetryTime = GlobalConfigMinimums.DownloadRetryTime
+	settingMap, ok := configPtr.AgentSettings[agentName]
+	if !ok {
+		// Agent Map not yet set. Create the map
+		settingMap := make(map[AgentSettingKey]ConfigItemValue)
+		configPtr.AgentSettings[agentName] = settingMap
 	}
-	if newgc.DomainBootRetryTime < GlobalConfigMinimums.DomainBootRetryTime {
-		log.Warnf("Enforce minimum DomainBootRetryTime received %d; using %d",
-			newgc.DomainBootRetryTime, GlobalConfigMinimums.DomainBootRetryTime)
-		newgc.DomainBootRetryTime = GlobalConfigMinimums.DomainBootRetryTime
+	settingMap[key] = configItemValue
+	return nil
+}
+
+func (configPtr *ConfigItemValueMap) SetGlobalValue(key GlobalSettingKey, newValue string) {
+	configPtr.GlobalSettings[key] = ConfigItemValue {
+		Key: string(key),
+		ItemType: ConfigItemTypeString,
+		stringValue: newValue,
 	}
-	if newgc.RktGCGracePeriod < GlobalConfigMinimums.RktGCGracePeriod {
-		log.Warnf("Enforce minimum RktGCGracePeriod received %d; using %d",
-			newgc.RktGCGracePeriod, GlobalConfigMinimums.RktGCGracePeriod)
-		newgc.RktGCGracePeriod = GlobalConfigMinimums.RktGCGracePeriod
+}
+
+func (configPtr *ConfigItemValueMap) DelAgentValue(key AgentSettingKey, agentName string)  {
+	settingMap, ok := configPtr.AgentSettings[agentName]
+	if !ok {
+		return
 	}
-	if newgc.Dom0MinDiskUsagePercent < GlobalConfigMinimums.Dom0MinDiskUsagePercent {
-		log.Warnf("Enforce minimum Dom0MinDiskUsagePercent received %d; using %d",
-			newgc.Dom0MinDiskUsagePercent, GlobalConfigMinimums.Dom0MinDiskUsagePercent)
-		newgc.Dom0MinDiskUsagePercent = GlobalConfigMinimums.Dom0MinDiskUsagePercent
+	delete(settingMap, key)
+	if len(settingMap) > 0 {
+		configPtr.AgentSettings[agentName] = settingMap
+	} else {
+		// No more settings for Agent.. So delete it from AgentSettings
+		delete(configPtr.AgentSettings, agentName)
 	}
-	return newgc
 }
+
+func (configPtr *ConfigItemValueMap) ResetGlobalValue(key GlobalSettingKey) {
+	specMap := NewConfigItemSpecMap()
+	configPtr.GlobalSettings[key] = specMap.GlobalSettings[key].DefaultValue()
+}
+
+func parseConfigItemToValue(spec ConfigItemSpec, item config.ConfigItem) (ConfigItemValue, error) {
```

Comment:
This should be a method of ConfigItemSpec
   spec.ParseValue(value string) (ConfigItemValue, error)

Isn't it??